hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -119,18 +119,18 @@ public class StoreFileInfo implements Configurable {
    */
   public StoreFileInfo(final Configuration conf, final FileSystem fs, final Path initialPath,
     final boolean primaryReplica) throws IOException {
-    this(conf, fs, null, initialPath, primaryReplica);
+    this(conf, fs, fs.getFileStatus(initialPath), primaryReplica);
```

Comment:
Should make sure constructor javadoc is updated as this is a Namenode call. If we call this often, we're incurring remote reads which could be a bad perf hit. Need to make sure current callers are behaving nicely (we _should_ be caching StoreFileInfo's already)

Reply:
I think this is intentional, we want to reduce the request to namenode...

Reply:
Ok, haven't thought about that. Let me revert this, and sanitize the path, instead.